### PR TITLE
Add extensions link in plugin page

### PR DESCRIPTION
**Pull request by AgnelFernando**
Fetch available plugin extensions from https://www.jenkins.io/doc/developer/extensions/contents.json
And use this list to check plugin has extensions in Plugin Page. If it has show link to plugin's extensions page.

issue link: https://github.com/jenkins-infra/plugin-site/issues/1131

### CODE REVIEW

**Comment by AgnelFernando** [comment]
Reply to https://github.com/jenkins-infra/jenkins.io/pull/5289#pullrequestreview-1036439150

Hi @MarkEWaite, This is how extension list is used in plugin site

**Review by halkeye — APPROVED** [review]
https://deploy-preview-1209--jenkins-plugin-site-pr.netlify.app/atlassian-jira-software-cloud/

Seems to work

Code is straightforward and readable

Lgtm

I'll give mark a chance to see it since you mentioned him then I'll merge

**Comment by zbynek** [comment]
I think "Extension Points" would be better label than just "Extensions".

Open question: would it make sense add a sub-section to the "Links" section for "Developer resources" to somehow indicate that Javadoc and Extension Points are not relevant for Jenkins admins, only for devs?

**Review by MarkEWaite — APPROVED** [review]
> I think "Extension Points" would be better label than just "Extensions".

I like the idea of "Extension Points" though I'll accept "Extensions" if others feel strongly that it should be "Extensions".

> Open question: would it make sense add a sub-section to the "Links" section for "Developer resources" to somehow indicate that Javadoc and Extension Points are not relevant for Jenkins admins, only for devs?

I'd rather not add another sub-section.  I think that most readers will see the words "Extension" and "Javadoc" and know that they are only interesting to those modifying plugins through Java.

File: plugins/plugin-site/src/templates/plugin.jsx [hunk 1]
```diff
@@ -176,6 +176,7 @@ function PluginPage({data: {jenkinsPlugin: plugin, reverseDependencies: reverseD
                         {plugin.scm && <div className="label-link"><a href={plugin.scm}>GitHub</a></div>}
                         <PluginIssueTrackers issueTrackers={plugin.issueTrackers} />
                         {plugin.hasPipelineSteps && <div className="label-link"><a href={`https://www.jenkins.io/doc/pipeline/steps/${plugin.name}`}>Pipeline Step Reference</a></div>}
+                        {plugin.hasExtensions && <div className="label-link"><a href={`https://www.jenkins.io/doc/developer/extensions/${plugin.name}`}>Extensions</a></div>}
```
Comment (zbynek) [inline plugins/plugin-site/src/templates/plugin.jsx]:
```suggestion
                        {plugin.hasExtensions && <div className="label-link"><a href={`https://www.jenkins.io/doc/developer/extensions/${plugin.name}`}>Extension Points</a></div>}
```

**Pull request merged**